model/trigger/payloads.go

```diff
@@ -0,0 +1,193 @@
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse body template")
+	}
+	buf := &bytes.Buffer{}
+	err = bodyTmpl.Execute(buf, t)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to execute email template")
+	}
+	body := buf.String()
+
+	subjectTmpl, err := template.New("subject").Parse(emailSubjectTemplate)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse subject template")
+	}
+	buf = &bytes.Buffer{}
+	err = subjectTmpl.Execute(buf, t)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to execute subject template")
+	}
+	subject := buf.String()
+
+	m := message.Email{
+		Subject:           subject,
+		Body:              body,
+		PlainTextContents: false,
+		Headers:           t.Headers,
+	}
+
+	return &m, nil
+}
+
+func webhookPayload(api restModel.Model, headers http.Header) (*util.EvergreenWebhook, error) {
+	bytes, err := json.Marshal(api)
+	if err != nil {
+		return nil, errors.Wrap(err, "error building json model")
+	}
+
+	return &util.EvergreenWebhook{
+		Body:    bytes,
+		Headers: headers,
+	}, nil
+}
+
+func jiraComment(t commonTemplateData) (*string, error) {
+	commentTmpl, err := ttemplate.New("jira-comment").Parse(jiraCommentTemplate)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse jira comment template")
+	}
+
+	buf := &bytes.Buffer{}
+	if err = commentTmpl.Execute(buf, t); err != nil {
+		return nil, errors.Wrap(err, "failed to make jira comment")
+	}
+	comment := buf.String()
+
+	return &comment, nil
+}
+
+func jiraIssue(t commonTemplateData) (*message.JiraIssue, error) {
+	const maxSummary = 254
+
+	comment, err := jiraComment(t)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to make jira issue")
+	}
+
+	issueTmpl, err := ttemplate.New("jira-issue").Parse(jiraIssueTitle)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse jira issue template")
+	}
+
+	buf := &bytes.Buffer{}
+	if err = issueTmpl.Execute(buf, t); err != nil {
+		return nil, errors.Wrap(err, "failed to make jira issue")
+	}
+	title, remainder := truncateString(buf.String(), maxSummary)
+	desc := *comment
+	if len(remainder) != 0 {
+		desc = fmt.Sprintf("...\n%s\n%s", remainder, desc)
+	}
+
+	issue := message.JiraIssue{
+		Summary:     title,
+		Description: desc,
+	}
+
+	return &issue, nil
+}
+
+// truncateString splits a string into two parts, with the following behavior:
+// If the entire string is <= capacity, it's returned unchanged.
+// Otherwise, the string is split at the (capacity-3)'th byte. The first string
+// returned will be this string + '...', and the second string will be the
+// remaining characters, if any
+func truncateString(s string, capacity int) (string, string) {
+	if len(s) <= capacity {
+		return s, ""
+	}
+	if capacity <= 0 {
+		return "", s
+	}
+
+	head := s[0:capacity-3] + "..."
+	tail := s[capacity-3:]
+
+	return head, tail
+}
+
+func slack(t commonTemplateData) (*notification.SlackPayload, error) {
```

Comment:
the name of this function should probably be slightly more verb-ey? makeSlackPayload.